Code/GraphMol/MarvinParse/MarvinDefs.cpp

```diff
@@ -63,11 +53,570 @@ ptree MarvinMolBase::toPtree() const {
+      std::string radical = atomArray.get<std::string>("<xmlattr>.radical", "");
+      boost::algorithm::split(radicals, radical, boost::algorithm::is_space());
+
+      std::vector<std::string> hydrogenCounts;
+      std::string hydrogenCount =
+          atomArray.get<std::string>("<xmlattr>.hydrogenCount", "");
+      boost::algorithm::split(hydrogenCounts, hydrogenCount,
+                              boost::algorithm::is_space());
+
+      std::vector<std::string> mrvValences;
+      std::string mrvValence =
+          atomArray.get<std::string>("<xmlattr>.mrvValence", "");
+      boost::algorithm::split(mrvValences, mrvValence,
+                              boost::algorithm::is_space());
+
+      std::vector<std::string> mrvAliases;
+      std::string mrvAlias =
+          atomArray.get<std::string>("<xmlattr>.mrvAlias", "");
+      boost::algorithm::split(mrvAliases, mrvAlias,
+                              boost::algorithm::is_space());
+
+      std::vector<std::string> rgroupRefs;
+      std::string rgroupRef =
+          atomArray.get<std::string>("<xmlattr>.rgroupRef", "");
+      boost::algorithm::split(rgroupRefs, rgroupRef,
+                              boost::algorithm::is_space());
+
+      std::vector<std::string> mrvStereoGroups;
+      std::string mrvStereoGroup =
+          atomArray.get<std::string>("<xmlattr>.mrvStereoGroup", "");
+      boost::algorithm::split(mrvStereoGroups, mrvStereoGroup,
+                              boost::algorithm::is_space());
+
+      std::vector<std::string> mrvMaps;
+      std::string mrvMap = atomArray.get<std::string>("<xmlattr>.mrvMap", "");
+      boost::algorithm::split(mrvMaps, mrvMap, boost::algorithm::is_space());
+
+      std::vector<std::string> sgroupRefs;
+      std::string sgroupRef =
+          atomArray.get<std::string>("<xmlattr>.sgroupRef", "");
+      boost::algorithm::split(sgroupRefs, sgroupRef,
+                              boost::algorithm::is_space());
+
+      std::vector<std::string> sgroupAttachmentPoints;
+      std::string sgroupAttachmentPoint =
+          atomArray.get<std::string>("<xmlattr>.sgroupAttachmentPoint", "");
+      boost::algorithm::split(sgroupAttachmentPoints, sgroupAttachmentPoint,
+                              boost::algorithm::is_space());
+
+      if (atomID != "") {
+        if (elementType == "") {
+          throw FileParseException(
+              "Expected an elementType array for an atomArray definition in MRV file");
+        }
+        if (elementTypes.size() < atomCount) {
+          throw FileParseException(
+              "There must be an element type for each atom id");
+        }
+      }
+
+      for (size_t i = 0; i < atomCount; ++i) {
+        auto *mrvAtom = new MarvinAtom();
+        this->pushOwnedAtom(mrvAtom);
+
+        this->atoms.push_back(mrvAtom);
+
+        mrvAtom->id = atomIds[i];
+
+        mrvAtom->elementType = elementTypes[i];
+
+        if (x2 != "" && y2 != "" && x2s.size() > i && y2s.size() > i) {
```

Comment:
Does that mean we can have atomCount > x2s.size() ?
Is that the way to say some atoms don't have coordinates?

Reply:
Some atoms may not have coords.  The list of coordinates can be shorter than the number of atoms.

Reply:
oh, what a fun file format...